client/js/components/statement/assessmentTable/DpSelectedItemsStatements.vue

```diff
@@ -173,15 +173,17 @@ export default {
     },
 
     selectionContainsUnclaimedFragments () {
-      return Object.keys(this.selectedElements)
-        .map(id => this.fragments[id] || [])
-        .map(storeFragment => storeFragment.fragments || [])
-        .reduce((acc, fragments) => {
-          return [...acc, ...fragments]
-        })
-        .filter(fragment => {
-          return (fragment.assignee.id !== this.currentUserId) || (fragment.departmentId && fragment.departmentId !== '')
-        }).length > 0
+      if (hasPermission('area_statements_fragment')) {
```

Comment:
Shouldn't it also work if in line 350 we put the permission check first? If it evaluates to false, this code should not be executed.
Should we keep the permission check here, I'd suggest to return `false` if it fails.

Reply:
I agree with you. I recognized it after pushing this solution

Reply:
I mean the return false I would like add here but the check I would like keep it here

Reply:
Could you explain why? :)

Reply:
No, I'd change the code in line 350 from

```
if (this.selectionContainsUnclaimedFragments && hasPermission('area_statements_fragment')) {
  this.triggerWarning('warning.edit.selection.fragments.not.claimed')
  isAllowed = false
}
```

to

```
if (hasPermission('area_statements_fragment') && this.selectionContainsUnclaimedFragments) {
  this.triggerWarning('warning.edit.selection.fragments.not.claimed')
  isAllowed = false
}
```

so in case the permission check fails, the computed is not even executed.